### PR TITLE
feat!: Upgrade to libbinaryen v110

### PR DESCRIPTION
**Pull request by phated**
This upgrades libbinaryen to v110 and adapts the bindings for the changes upstream.

I separated each thing into their own commits so it will be easier for @ospencer to review.

### CODE REVIEW

File: src/binaryen_stubs_expressions.js [hunk 1]
```diff
@@ -173,16 +173,89 @@ function caml_binaryen_global_set(wasm_mod, name, value) {
...
-    bytes,
-    signed,
-    offset,
-    align,
-    typ,
-    ptr
-  );
+//Requires: caml_jsstring_of_string
+//Requires: caml_failwith
+function caml_binaryen_load(
```
Comment (phated) [inline src/binaryen_stubs_expressions.js:178]: @ospencer please don't kill me for this implementation

File: src/binaryen_stubs_expressions.js [hunk 2]
```diff
@@ -194,22 +267,72 @@ function caml_binaryen_load__bytecode() {
...
-    bytes,
-    offset,
-    align,
-    ptr,
-    value,
-    typ
-  );
+//Requires: caml_jsstring_of_string
+//Requires: caml_failwith
+function caml_binaryen_store(
```
Comment (phated) [inline src/binaryen_stubs_expressions.js:279]: @ospencer please don't kill me for this implementation

**Review by ospencer — APPROVED** [review]
Overall looks really good, just one thought on the load functions.

Up to you if you want to make the changes but it's fine either way.

Great work!

**Comment by phated** [comment]
Updated, but waiting on https://github.com/ocaml/opam-repository/pull/22863 to be merged so we can release these breaking updates

**Pull request merged**